lib/modules/datasource/maven/util.ts

```diff
@@ -334,10 +379,12 @@ export async function downloadMavenXml(
   }
 
   if (protocol === 's3:') {
-    const res = await downloadS3Protocol(pkgUrl);
-    if (res) {
-      return { xml: new XmlDocument(res) };
-    }
+    const rawResult = await downloadS3Protocol(pkgUrl);
+    const xmlResult = rawResult.transform(({ isCacheable, data }): MavenXml => {
+      const xml = new XmlDocument(data);
+      return { xml };
+    });
+    return xmlResult.unwrapOr({});
```

Comment:
can we catch and log XML parse errors?

Reply:
Will be done in future, when some function will combine both http and s3